internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -1718,75 +1712,53 @@ func TestComputeMachineDeployment(t *testing.T) {
-		}).
-		WithStatusFields(map[string]interface{}{
-			"status.version": "v1.2.1",
-		}).
-		Build()
-	controlPlaneScaling := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version":  "v1.2.3",
-			"spec.replicas": int64(2),
-		}).
-		WithStatusFields(map[string]interface{}{
-			"status.version":             "v1.2.3",
-			"status.replicas":            int64(1),
-			"status.updatedReplicas":     int64(1),
-			"status.readyReplicas":       int64(1),
-			"status.unavailableReplicas": int64(0),
-		}).
-		Build()
-	controlPlaneDesired := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version": "v1.2.3",
-		}).
+	controlPlaneObj := builder.ControlPlane("test1", "cp1").
 		Build()
 
+	mdName := "md-1"
+	currentMachineDeploymentState := &scope.MachineDeploymentState{Object: builder.MachineDeployment("test1", mdName).WithVersion("v1.2.2").Build()}
+
 	tests := []struct {
-		name                          string
-		machineDeploymentTopology     clusterv1.MachineDeploymentTopology
-		currentMachineDeploymentState *scope.MachineDeploymentState
-		upgradingMachineDeployments   []string
-		upgradeConcurrency            int
-		currentControlPlane           *unstructured.Unstructured
-		desiredControlPlane           *unstructured.Unstructured
-		topologyVersion               string
-		expectedVersion               string
+		name                                 string
+		machineDeploymentTopology            clusterv1.MachineDeploymentTopology
+		currentMachineDeploymentState        *scope.MachineDeploymentState
+		upgradingMachineDeployments          []string
+		upgradeConcurrency                   int
+		controlPlaneStartingUpgrade          bool
+		controlPlaneUpgrading                bool
+		controlPlaneScaling                  bool
+		controlPlaneProvisioning             bool
+		afterControlPlaneUpgradeHookBlocking bool
+		topologyVersion                      string
+		expectedVersion                      string
+		expectPendingCreate                  bool
+		expectPendingUpgrade                 bool
 	}{
 		{
 			name:                          "should return cluster.spec.topology.version if creating a new machine deployment",
 			currentMachineDeploymentState: nil,
 			topologyVersion:               "v1.2.3",
 			expectedVersion:               "v1.2.3",
+			expectPendingCreate:           false,
+		},
+		{
+			name:                "should return cluster.spec.topology.version if creating a new machine deployment and if control plane is not stable - marked as pending create",
+			controlPlaneScaling: true,
+			machineDeploymentTopology: clusterv1.MachineDeploymentTopology{
+				Name: "md-topology-1",
+			},
+			topologyVersion:     "v1.2.3",
+			expectedVersion:     "v1.2.3",
+			expectPendingCreate: true,
+		},
+		{
+			name: "should return cluster.spec.topology.version if creating a new machine deployment and if control plane is stable - not marked as pending create",
```

Comment:
I'm not sure if I understand how this and the first case make sense. Is one of the test cases incorrectly set up?

(wondering how we can create a MD without a machineDeploymentTopology)

Reply:
Good catch! The first case was incorrectly setup and the second one is correctly setup. The first case still passed because md topology name in the first case would be `""` and since `expectPendingCreate: false` in the first case we are checking that `""` does not exist in the pending create set (verify it should not be marked as pending create) which it wont.

Does this answer your question?

Reply:
Yup. Looks like we now have 2 identical test cases?

Reply:
They are not identical. In one case the control plane is stable so `expectPendingCreate` is `false` and in the second case the control plane is scaling (not stable) so `expectPendingCreate`  is `true`.

Reply:
Not those 2. Those:
```go
		{
			name:                          "should return cluster.spec.topology.version if creating a new machine deployment and if control plane is stable",
			currentMachineDeploymentState: nil,
			machineDeploymentTopology: clusterv1.MachineDeploymentTopology{
				Name: "md-topology-1",
			},
			topologyVersion:     "v1.2.3",
			expectedVersion:     "v1.2.3",
			expectPendingCreate: false,
		},
```

```
		{
			name: "should return cluster.spec.topology.version if creating a new machine deployment and if control plane is stable - not marked as pending create",
			machineDeploymentTopology: clusterv1.MachineDeploymentTopology{
				Name: "md-topology-1",
			},
			topologyVersion:     "v1.2.3",
			expectedVersion:     "v1.2.3",
			expectPendingCreate: false,
		},
```

Reply:
Ah!  you are right.